lib/spack/spack/package.py

```diff
@@ -779,6 +779,15 @@ def cleanup():
                          "Manually remove this directory to fix:",
                          self.prefix)
 
+            if not (keep_prefix and keep_stage):
+                self.do_clean()
+            else:
+                tty.warn("Keeping stage in place despite error.",
+                         "Spack will refuse to uninstall dependencies of this package." +
+                         "Manually remove this directory to fix:",
+                         self.stage.path)
```

Comment:
I should look at this in more detail.  Having the *stage* in place shouldn't cause Spack to refuse to uninstall dependencies... does it?  Having the *prefix* in place should, though.